bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/ModuleUtil.java

```diff
@@ -0,0 +1,578 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.automation.Module;
+import org.eclipse.smarthome.automation.core.type.ModuleTypeManager;
+import org.eclipse.smarthome.automation.type.ActionType;
+import org.eclipse.smarthome.automation.type.ConditionType;
+import org.eclipse.smarthome.automation.type.Input;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.Output;
+import org.eclipse.smarthome.automation.type.TriggerType;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Util used for resolving references in {@link ConfigDescriptionParameter} , {@link Input} and {@link Output} in Module
+ *
+ * @author Vasil Ilchev - Initial Contribution
+ */
+public class ModuleUtil {
+    /**
+     * Input/Output <code>reference</code> and Config property <code>context</code> can start with this symbol.
+     * i.e. That indicates that Config property points to other Config property.
+     */
+    public static final String PARSABLE_DOLLAR = "$";
+    /**
+     * ModuleType UID separator.
+     * i.e. SystemModuleType:CustomModuleType
+     */
+    public static final String MODULE_SEPARATOR = ":";
+
+    private static Logger log;
```

Comment:
it is a kind of lazy initialization - Logger instance is created on first error or warning